### PR TITLE
Add Astra DB VectorStore functionality

### PR DESCRIPTION
**Pull request by erichare**
# Description

This adds an `AstraDBVectorStore` class, built upon the `astrapy` client, to support using Astra DB as a backend.

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] I stared at the code and made sure it makes sense
- [x] I have added tests that prove my fix is effective or that my feature works

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama_index/vector_stores/astra.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+            raise ImportError(import_err_msg)
+
+        # Set all the required class parameters
+        self._embedding_dimension = embedding_dimension
+        self._ttl_seconds = ttl_seconds
+
+        _logger.debug("Creating the Astra DB table")
+
+        # Build the Astra DB object
+        self.astra_db = AstraDB(
```
Comment (nerdai) [inline llama_index/vector_stores/astra.py:74]: very nit: can we make these "private" `_astra_db`. I think we've been generally trying to adopt this standard throughout all of our vector store integrations (though some have slipped thru the cracks it seems)

File: llama_index/vector_stores/astra.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+            api_endpoint=api_endpoint, token=token, namespace=namespace
+        )
+
+        # Create the new collection
+        self.astra_db.create_collection(
+            collection_name=collection_name, dimension=embedding_dimension
+        )
+
+        # Connect to the newly created collection
+        self.astra_db_collection = AstraDBCollection(
```
Comment (nerdai) [inline llama_index/vector_stores/astra.py:84]: same nit comment as before

**Review by nerdai — APPROVED** [review]
Thanks @erichare! This looks overall pretty good.

Is it possible to add a unit test for this?

File: llama_index/vector_stores/astra.py (hunk 1 above)
Comment (erichare) [reply llama_index/vector_stores/astra.py]: Done! Thanks!

**Comment by nerdai** [comment]
Thanks @erichare! Glad to see we got over the line with the lint check!

It all LGTM, but before merging, could we actually do a few more things.

- add entry in `CHANGELOG.md` under `Unreleased` >> `New Features`
- add AstraDB to the table found in `docs/module_guides/storing/vector_stores.md`
- also if you want to add AstraDB to the list in `docs/community/integrations/vector_stores.md` linking to some quickstart or installation guides

**Comment by erichare** [comment]
> Thanks @erichare! Glad to see we got over the line with the lint check!
> 
> It all LGTM, but before merging, could we actually do a few more things.
> 
> * add entry in `CHANGELOG.md` under `Unreleased` >> `New Features`
> * add AstraDB to the table found in `docs/module_guides/storing/vector_stores.md`
> * also if you want to add AstraDB to the list in `docs/community/integrations/vector_stores.md` linking to some quickstart or installation guides

Thanks so much! I think i added all of this, hopefully correctly :)

**Comment by nerdai** [comment]
> > Thanks @erichare! Glad to see we got over the line with the lint check!
> > It all LGTM, but before merging, could we actually do a few more things.
> > 
> > * add entry in `CHANGELOG.md` under `Unreleased` >> `New Features`
> > * add AstraDB to the table found in `docs/module_guides/storing/vector_stores.md`
> > * also if you want to add AstraDB to the list in `docs/community/integrations/vector_stores.md` linking to some quickstart or installation guides
> 
> Thanks so much! I think i added all of this, hopefully correctly :)

I think you did and it looks fabulous! Thanks so much for the contribution :). Merging now.

**Pull request merged**